demisto_sdk/commands/unify/generic_module_unifier.py

```diff
@@ -0,0 +1,115 @@
+
+import json
+import os
+import sys
+from argparse import FileType
+from typing import Dict, Optional
+
+import click
+
+from demisto_sdk.commands.common.constants import FileType
+from demisto_sdk.commands.common.tools import get_pack_name, print_error, find_type
+
+
+class GenericModuleUnifier:
+    """
+    Unifies a GenericModule object with it's Dashboards
+    """
+
+    def __init__(self, input: str, output: str = '', force: bool = False):
+        """
+        Init a GenericModuleUnifier
+        Args:
+            input: a path of the GenericModule file to unify.
+            output: The output dir to write the unified GenericModule json to.
+            force: if True - Forcefully overwrites the preexisting unified GenericModule file if one exists.
+        """
+
+        if find_type(input) == FileType.GENERIC_MODULE:
+            self.input_path = input
```

Comment:
Isn't this checked in `__main__`?

Reply:
My rationale was to add this check again inside the constructor to handle cases of calling it from another function (not main function). But you're right that right now it's unnecessary. I'm removing it.